utils/model_uploader/sparse_model_autotracing.py

```diff
@@ -0,0 +1,465 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import argparse
+import json
+import os
+import shutil
+import sys
+from typing import Optional, Tuple
+
+THIS_DIR = os.path.dirname(__file__)
+ROOT_DIR = os.path.join(THIS_DIR, "../..")
+sys.path.append(ROOT_DIR)
+
+from huggingface_hub import HfApi
+
+from opensearch_py_ml.ml_commons import MLCommonClient
+from opensearch_py_ml.ml_models import SparseEncodingModel
+from tests import OPENSEARCH_TEST_CLIENT
+from utils.model_uploader import autotracing_utils
+
+BOTH_FORMAT = "BOTH"
+TORCH_SCRIPT_FORMAT = "TORCH_SCRIPT"
+ONNX_FORMAT = "ONNX"
+
+TEMP_MODEL_PATH = "temp_model_path"
+TORCHSCRIPT_FOLDER_PATH = "model-torchscript/"
+ONNX_FOLDER_PATH = "model-onnx/"
+UPLOAD_FOLDER_PATH = "upload/"
+MODEL_CONFIG_FILE_NAME = "ml-commons_model_config.json"
+OUTPUT_DIR = "trace_output/"
+LICENSE_VAR_FILE = "apache_verified.txt"
+DESCRIPTION_VAR_FILE = "description.txt"
+TEST_SENTENCES = ["Nice to meet you.", "I like playing football.", "Thanks."]
+RTOL_TEST = 1e-03
+ATOL_TEST = 1e-05
+
+
+def verify_license_by_Hf(model_id: str):
+    api = HfApi()
+    model_info = api.model_info(model_id)
+    model_license = model_info.cardData.get("license", "License information not found.")
+    if model_license == "apache-2.0":
+        return True
+    else:
+        return False
+
+
+def trace_sparse_encoding_model(
+    model_id: str,
+    model_version: str,
+    model_format: str,
+    model_description: Optional[str] = None,
+) -> Tuple[str, str]:
```

Comment:
This method is quite similar to trace_sentence_transformer_model in model_autotracing.py. Can you try to refactor the code for code reusage?

Reply:
+1